rest_framework_json_api/utils.py

```diff
@@ -394,9 +411,14 @@ def format_drf_errors(response, context, exc):
             errors.extend(format_error_object(message, "/data", response))
     # handle all errors thrown from serializers
     else:
+        serializer = get_serializer_from_context(context)
```

Comment:
I assume you added this complex method `get_serializer_from_context` because of `test_errors.py`?

I have a feeling the test setup in `test_errors.py` is not correct as an `APIView` does not have a serializer and it just fakes to have one. Basically only views which derive from `rest_framework.generics.GenericAPIView` have serializer support. And then a simple call to `view.get_serializer()` will do.

The question remains though what do we do with views which directly derive from `APIView`?. We cannot correctly determine whether it is a relationship or an attribute. In the [JSON:API spec](https://jsonapi.org/format/#errors) pointer is optional.

So I guess the best strategy will be if the view does not derive from `GeneriAPIView` then simply set `pointer` to `None`.